draft-ietf-wimse-s2s-protocol.md

```diff
@@ -486,25 +486,21 @@ A signed response would be:
 
 # Using Mutual TLS for Service To Service Authentication {#mutual-tls}
 
-The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjectAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  The WIMSE certificate may contain SubjectAltName extensions of other types such as DNSName.
+The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjectAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  If the workload will act as a TLS server for clients that do not understand WIMSE workload identities it is RECOMMENDED that WIMSE certificate contain a SubjectAltName of DNSName with the appropriate DNS names for the server. The certificate may contain SubjectAltName extensions of other types.
 
-WIMSE certificates may be used to secure both server and client connections.  When validating a WIMSE certificate, the relying party MUST validate that the CA issuer for the WIMSE certificate is authorized to issue certificates for the trust domain of the WIMSE workload identified by the certificate. Other PKIX path validation rules apply.
+WIMSE certificates may be used to authenticate both the server and client side of the connections.  When validating a WIMSE certificate, the relying party MUST use the trust anchors configured for the trust domain in the WIMSE identity to validate the peer's certificate.  Other PKIX {{!RFC5280}} path validation rules apply. WIMSE clients and servers MUST validate that the trust domain portion of the WIMSE certificate matches the expected trust domain for the other side of the connection.
 
 Servers wishing to use the WIMSE certificate for authorizing the client MUST require client certificate authentication in the TLS handshake. Other methods of post handshake authentication are not specified by this document.
 
-WIMSE clients and servers MUST validate that the trust domain portion of the WIMSE certificate matches the expected trust domain for the other side of the connection.
+WIMSE server certificates MUST have the id-kp-serverAuth extended key usage {{!RFC5280}} field set and WIMSE client certificates MUST have the id-kp-clientAuth extended key usage field set. A certificate that is used for both client and server connections may has both fields set. This specification does not make any other requirements beyond {{!RFC5280}} on the contents of WIMSE certificates or on the certificates that issue WIMSE certificates.
```

Comment:
Asking out of ignorance: in the past, EKU was hit-and-miss. Is it now implemented so widely that we can have it as a MUST?

Reply:
I think its probably still a little hit and miss.  I think we can try to raise the bar here, but maybe its not really all that important enough to make it a barrier. I'd be happy with a SHOULD.

Reply:
Resolved by Yaron's suggestion.